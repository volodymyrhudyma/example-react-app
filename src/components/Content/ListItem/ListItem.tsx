import React, { Component } from 'react';
import {
  Wrapper,
  CardInner,
  UserWrapper,
  AvatarWrapper,
  DetailsWrapper,
  FullName,
  Email,
  Position,
  TeaserWrapper,
  TitleWrapper,
  TextWrapper,
  TagsWrapper,
  ActionsWrapper,
  DeleteWrapper,
  ReadWrapper,
  ReadMarkWrapper,
  LimitWidth,
  EndWrapper,
} from './styled/ListItem.styled';
import Card from 'src/ui-kit/Card';
import Avatar from 'src/ui-kit/Avatar';
import icons from 'src/constants/icons';
import Title from 'src/ui-kit/Title';
import Icon from 'src/ui-kit/Icon';
import Button from 'src/ui-kit/Button';
import Tag from 'src/ui-kit/Tag';
import Dot from 'src/ui-kit/Dot';
import { CARD } from 'src/constants/layout';
import { IArticle } from 'src/types';
import { WithTranslation } from 'react-i18next';

interface IListItemProps extends WithTranslation {
  layout: string;
  article: IArticle;
  readArticle: (article: IArticle) => void;
  deleteArticle: (id: number) => void;
}

class ListItem extends Component<IListItemProps> {
  readArticle = () => {
    const { readArticle, article } = this.props;
    readArticle(article);
  };
  deleteArticle = () => {
    const { deleteArticle, article } = this.props;
    deleteArticle(article.id);
  };
  render() {
    const { t, layout, article } = this.props;
    const isCard = layout === CARD;
    const isArticleRead = article.isRead;
    return (
      <Wrapper isCard={isCard}>
        <Card
          padding={isCard ? '3rem' : '1.5rem'}
          width={isCard ? '300px' : 'auto'}
          height={isCard ? '100%' : 'auto'}
        >
          <CardInner isCard={isCard}>
            <UserWrapper isCard={isCard}>
              <AvatarWrapper>
                <Avatar src={icons.articleAvatar} />
              </AvatarWrapper>
              <DetailsWrapper isCard={isCard}>
                <FullName>
                  <Title type="h4">
                    <LimitWidth>{article.author}</LimitWidth>
                  </Title>
                </FullName>
                <Email>{article.email}</Email>
                <Position>{article.jobTitle}r</Position>
              </DetailsWrapper>
            </UserWrapper>
            <TeaserWrapper isCard={isCard}>
              <TitleWrapper isCard={isCard}>
                <Title type="h4">{article.title}</Title>
              </TitleWrapper>
              <TextWrapper>{article.desc}</TextWrapper>
            </TeaserWrapper>
            <EndWrapper>
              <TagsWrapper isCard={isCard}>
                {article.tags &&
                  article.tags.map(tag => <Tag key={tag}>{tag}</Tag>)}
              </TagsWrapper>
              <ActionsWrapper isCard={isCard}>
                <DeleteWrapper onClick={this.deleteArticle}>
                  <Icon src={icons.delete} />
                </DeleteWrapper>
                {!isArticleRead && (
                  <ReadWrapper>
                    <Button onClick={this.readArticle}>
                      {t('articles.readOne')}
                    </Button>
                  </ReadWrapper>
                )}
              </ActionsWrapper>
            </EndWrapper>
            {!isArticleRead && (
              <ReadMarkWrapper>
                <Dot />
              </ReadMarkWrapper>
            )}
          </CardInner>
        </Card>
      </Wrapper>
    );
  }
}

export default ListItem;
