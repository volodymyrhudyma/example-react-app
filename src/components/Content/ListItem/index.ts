import { connect } from 'react-redux';
import { IAppState, IArticle } from 'src/types';
import Component from './ListItem';
import { readArticle, deleteArticle } from 'src/actions/article';
import { withTranslation } from 'react-i18next';

const mapStateToProps = (state: IAppState) => ({
  layout: state.layout.layout,
});

const mapDispatchToProps = (dispatch: any) => {
  return {
    readArticle: (article: IArticle) => dispatch(readArticle(article)),
    deleteArticle: (id: number) => dispatch(deleteArticle(id)),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(Component));
