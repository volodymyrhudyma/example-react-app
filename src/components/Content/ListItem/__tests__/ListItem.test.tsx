import { shallow } from 'enzyme';
import React from 'react';
import ListItem from '../ListItem';
import i18n from 'i18next';

describe('ListItem with Enzyme and Jest', () => {
  it('should render correctly', () => {
    const component = shallow(
      <ListItem
        t={jest.fn()}
        i18n={i18n}
        tReady={false}
        layout="TABLE"
        article={{
          avatar: '',
          author: '',
          email: '',
          isRead: false,
          jobTitle: '',
          title: '',
          desc: '',
          tags: [],
          id: 0,
        }}
        deleteArticle={jest.fn()}
        readArticle={jest.fn()}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
