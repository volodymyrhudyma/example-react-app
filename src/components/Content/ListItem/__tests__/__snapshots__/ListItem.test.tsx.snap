// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem with Enzyme and Jest should render correctly 1`] = `
<styled.div
  isCard={false}
>
  <Card
    height="auto"
    padding="1.5rem"
    width="auto"
  >
    <styled.div
      isCard={false}
    >
      <styled.div
        isCard={false}
      >
        <styled.div>
          <Avatar
            height={60}
            src="/icons/avatar-article.png"
            width={60}
          />
        </styled.div>
        <styled.div
          isCard={false}
        >
          <styled.div>
            <Title
              type="h4"
            >
              <styled.div />
            </Title>
          </styled.div>
          <styled.div />
          <styled.div>
            r
          </styled.div>
        </styled.div>
      </styled.div>
      <styled.div
        isCard={false}
      >
        <styled.div
          isCard={false}
        >
          <Title
            type="h4"
          />
        </styled.div>
        <styled.div />
      </styled.div>
      <styled.div>
        <styled.div
          isCard={false}
        />
        <styled.div
          isCard={false}
        >
          <styled.div
            onClick={[Function]}
          >
            <Icon
              alt="Icon"
              height={24}
              src="/icons/delete.svg"
              width={24}
            />
          </styled.div>
          <styled.div>
            <Button
              onClick={[Function]}
            />
          </styled.div>
        </styled.div>
      </styled.div>
      <styled.div>
        <Dot />
      </styled.div>
    </styled.div>
  </Card>
</styled.div>
`;
